subprojects/kotlin-dsl-provider-plugins/src/main/kotlin/org/gradle/kotlin/dsl/provider/plugins/precompiled/PrecompiledScriptPlugin.kt

```diff
@@ -139,16 +139,18 @@ private
 fun packageNameOf(code: String): String? =
     KotlinLexer().run {
         start(code)
-        skipWhiteSpaceAndComments()
-        when (tokenType) {
-            KtTokens.PACKAGE_KEYWORD -> {
-                advance()
-                skipWhiteSpaceAndComments()
-                parseQualifiedName()
-            }
 
-            else -> null
+        while (tokenType != null && tokenType != KtTokens.PACKAGE_KEYWORD) {
```

Comment:
This makes a change in behavior in that before we would abort lexing early, now we need to lex the whole script in the absence of a `package` statement. This might have a bad performance impact and could cause error cases to change and report less useful diagnostics.

Reply:
Addressed in later commits.